plugins/source/aws/client/services.go

```diff
@@ -831,6 +833,18 @@ type SecretsManagerClient interface {
 	ListSecrets(ctx context.Context, params *secretsmanager.ListSecretsInput, optFns ...func(*secretsmanager.Options)) (*secretsmanager.ListSecretsOutput, error)
 }
 
+//go:generate mockgen -package=mocks -destination=./mocks/mock_service_catalog.go . ServiceCatalogClient
+type ServiceCatalogClient interface {
+	ListPortfolios(ctx context.Context, input *servicecatalog.ListPortfoliosInput, optFns ...func(*servicecatalog.Options)) (*servicecatalog.ListPortfoliosOutput, error)
+	SearchProductsAsAdmin(ctx context.Context, params *servicecatalog.SearchProductsAsAdminInput, optFns ...func(*servicecatalog.Options)) (*servicecatalog.SearchProductsAsAdminOutput, error)
+	SearchProvisionedProducts(ctx context.Context, params *servicecatalog.SearchProvisionedProductsInput, optFns ...func(*servicecatalog.Options)) (*servicecatalog.SearchProvisionedProductsOutput, error)
+}
+
+//go:generate mockgen -package=mocks -destination=./mocks/mock_service_catalog_app_registry.go . ServiceCatalogAppRegistryClient
+type ServiceCatalogAppRegistryClient interface {
+	ListTagsForResource(ctx context.Context, params *servicecatalogappregistry.ListTagsForResourceInput, optFns ...func(*servicecatalogappregistry.Options)) (*servicecatalogappregistry.ListTagsForResourceOutput, error)
```

Comment:
This is used to get tags for `ListPortfolios` and `SearchProductsAsAdmin` results. Completely unsure if it will work or error out.